.github/workflows/rust.yml

```diff
@@ -222,3 +218,36 @@ jobs:
 
       - name: Check hello_world
         run: cargo check -p hello_world
+
+  # ---------------------------------------------------------------------------
+
+  tests:
+    name: Run tests
+    # We run the tests on macOS because it will run with a actual GPU
+    runs-on: macos-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          lfs: true
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: 1.76.0
+
+      - name: Set up cargo cache
+        uses: Swatinem/rust-cache@v2
+
+      - name: Run tests
+        # TODO(lucasmerlin): Enable --all-features (currently this breaks the rendering in the tests because of the `unity` feature)
+        run: cargo test
+
+      - name: Run doc-tests
+        # TODO(lucasmerlin): Enable --all-features (currently this breaks the rendering in the tests because of the `unity` feature)
+        run: cargo test --doc
```

Comment:
@emilk the `unity` feature completely breaks rendering with egui_wgpu if enabled, so we can't use --all-features anymore. Do you have an idea how we could work around that? 
My only idea is to manually enable all features except the unity one but I really don't like that.

Or we add a `no_unity` feature that overwrites the unity feature but that seems terrible as well.